apps/processing/src/services/sharedDependencies.service.ts

```diff
@@ -24,9 +20,9 @@ import { Environment } from "../config/index.js";
 
 export type SharedDependencies = {
     core: Omit<CoreDependencies, "evmProvider">;
-    registries: {
-        eventsRegistry: IEventsRegistry;
-        strategyRegistry: IStrategyRegistry;
+    registriesRepositories: {
+        eventRegistryRepository: IEventRegistryRepository;
+        strategyRegistryRepository: IStrategyRegistryRepository;
```

Comment:
i think the previous way was ok. I...Registry should be de dependency not I...RegistryRepository. Lmk if iam missing something

Reply:
i've changed the way it works by creating an instance per chain but using the same repository, since it's not really necessary to load into memory the `lastProcessedEvent` of another chain (if we deploy a container per chain, this is more obvious to see).

if this makes sense, the same change applies for strategies (we create an instance per chain) but didn't want to mix and make the change for StrategyRegistry too. if we go for this approach, i'll make the change in another PR

if smth it's not clear, we can discuss it offline 😄